# bioinformatics 
A repository for my bioinformatics tools
<br><br>
## Description
This Github repository was created to store scripts and programs that I have written to aid my bioinformatics research.
<br><br>
## Getting Started
### Dependencies
* git
* Python 3
* pip3 ([View requirements.txt](https://github.com/bbaggins2005/bioinformatics/blob/main/requirements.txt))
<br><br>
### Installation Procedure

#### 1) Clone Bioinformatics from github
```bash
git clone https://github.com/bbaggins2005/bioinformatics.git
cd bioinformatics 
```

#### 2) Create Python virtual environment
```bash
python -m venv .venv
source .venv/bin/activate  # for linux
.venv\Scripts\activate.bat # for windows
```

#### 3) Install required Python packages
```bash
pip install -r  requirements.txt
```

#### 4) Upgrade PyVCF3
```bash
pip install --upgrade PyVCF3
```

#### 5) List installed packages, including editables.
```bash
pip list
```

#### 6) Remediate [PyVCF3 Issue #6](https://github.com/dridk/PyVCF3/issues/6)
* click [here](https://github.com/bbaggins2005/bioinformatics/blob/main/docs/PyVCF3_Issue6.md)

<br><br>
## Authors and Contributors
Reece Chae
<br><br>
## References and Helpful Links
* [BCFtools](https://github.com/samtools/bcftools): utilities for variant calling and manipulating VCFs and BCFs.
* [vcftools](https://github.com/vcftools/vcftools): A set of tools written in Perl and C++ for working with VCF files, such as those generated by the 1000 Genomes Project.
* [Ensembl Variant Effect Predictor](https://github.com/Ensembl/ensembl-vep): predicts the functional effects of genomic variant and filters annotations.
* [PyVCF3](https://github.com/dridk/PyVCF3): A Variant Call Format reader for Python.